src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/AppModel/AssemblyFilter.cs

```diff
@@ -43,13 +43,12 @@ internal void FilterCallback(Object sender, AssemblyLoadEventArgs args)
         //appends assembly name with file version to generate a unique entry for the assembly lookup process
         private string AssemblyNameWithFileVersion(Assembly a)
         {
-            FileVersionInfo fileVersionInfo;
             StringBuilder sb = new StringBuilder(a.FullName);
 
-            fileVersionInfo = FileVersionInfo.GetVersionInfo(a.Location);
-            if (fileVersionInfo != null && fileVersionInfo.ProductVersion != null)
+            Version assemblyVersion = a.GetName().Version;
```

Comment:
Another case where the two methods of getting a version string produce different results.  This one matters.  ProductVersion can be different from FileVersion, and presumably there's a reason the old code uses ProductVersion.  The new code breaks that scenario.

Again, please use the old code, with a fallback to the new code if running as single-file.  And find out what breaks by not using ProductVersion here, so that we can warn people not to use single-file if they need that scenario.

Reply:
For this case I end up using the same code path that FileVersionInfo uses in linux to get the ProductVersion, by default is the assembly.version and then if there is an InformationalVersionAttribute in the assembly it gets changed to that value. I'm not fully sure is the same in windows because it uses COM and is difficult to read, but I leave this logic only for the case in which the assembly.location is empty